compiler/modules/comm.diagnostic/diagnostic.bal

```diff
@@ -112,7 +113,54 @@ public function messageFormat(Template t) returns string {
 
 public function format(Diagnostic d) returns string[] {
     Location loc = d.location;
+    string [] errMessage = [];
+    string line = loc.file.filename();
     LineColumn lc = locationLineColumn(loc);
     // lineColumn returns 0-based lines currently
-    return [loc.file.filename() + ":" + lc[0].toString() + ":" + (lc[1] + 1).toString() + ": error: " + d.message];
+    line += ":" + lc[0].toString() + ":" + (lc[1] + 1).toString();
+    line += ": error: " + d.message;
+    errMessage.push(line);
+    Range|Position range = loc.range;
+    errMessage.push(...fileLines(loc.file, range));
+    return errMessage;
+}
+
+function fileLines(File file, Range|Position range) returns string[] {
+    string[] lines = file.getRange(range);
```

Comment:
I just want to print out one line at this stage (I think that's one LLVM does) - the one containing the start of the range. We should generally aim to ensure that the primary range does not span multiple lines.  Multiple lines will need a fancier display.

Reply:
fixed